lib/assets/javascripts/utils/ariatiseForm.js

```diff
@@ -0,0 +1,122 @@
+
+    For example the following form :
+    <form>
+        <div class="form-group">
+            <label class="control-label" for="name">Name</label>
+            <input type="text" class="form-control" id="name" aria-required="true">
+        </div>
+        <div class="form-group">
+            <label class="control-label" for="email">Email</label>
+            <input type="email" class="form-control" id="email" aria-required="true">
+        </div>
+        <div class="form-group">
+            <label class="control-label" for="name">Subject</label>
+            <input type="text" class="form-control" id="subject" aria-required="false">
+        </div>
+        <button name="button" type="submit" class="btn btn-default">Submit</button>
+    </form>
+    will be augmentated as follows:
+    <form>
+        <div class="form-group">
+            <label class="control-label" for="name">Name</label>
+            <input type="text" class="form-control" id="name" aria-required="true" aria-describedby="help0">
+            <span id="help0" class="help-block" style="display:none;">Please fill out this field with a valid text.</span>
+        </div>
+        <div class="form-group">
+            <label class="control-label" for="email">Email</label>
+            <input type="email" class="form-control" id="email" aria-required="true" aria-describedby="help1">
+            <span id="help1" class="help-block" style="display:none;">Please fill out this field with a valid email.</span>
+        </div>
+        <div class="form-group">
+            <label class="control-label" for="name">Subject</label>
+            <input type="text" class="form-control" id="subject" aria-required="false">
+        </div>
+        <button name="button" type="submit" class="btn btn-default">Submit</button>
+    </form>
+    and any time the buttton is clicked the validation according to each type (e.g. text, email) will be triggered. An invalid result for a form-control will:
+    1. Add has-error class to its form-group parent and aria-invalid="true" to the form-control
+    2. Show its help-block following sibling
+    3. Prevent form to be submitted
+*/
+(function(ctx){
+    var requiredFields=(function(selector){
+        return $(selector).find('.form-control').filter('[aria-required="true"]');
+    });
+    var blockHelp=(function (id,type){
+        var msg='Please fill out this field with a valid '+type+'.';    //TODO internationalisation
+        return '<span id="'+id+'" class="help-block" style="display:none;">'+msg+'</span>';
+    });
+    var ariaDescribedBy=(function(value){
+        return { 'aria-describedby': value };
+    });
+    var ariaInvalid=(function(value){
+        return { 'aria-invalid': value };
+    });
+    var validationStates={
+            hasWarning: 'has-warning',
+            hasError: 'has-error',
+            hasSuccess: 'has-success'
+    };
+    var getTypeForSubmittableElement=(function(el){
```

Comment:
This is good. I like that it is generic. We may want to refactor later to allow for custom error messages, maybe setting a specific attr on the input like 'dmproadmap-error'. This would allow us to provide messaging like 'Passwords must be t least 8 characters and contain ...'